src/Google.Bigquery.V2/BigqueryClient.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Bigquery.v2;
+using Google.Apis.Bigquery.v2.Data;
+using Google.Apis.Services;
+using System.Threading.Tasks;
+
+namespace Google.Bigquery.V2
+{
+    /// <summary>
+    /// Abstract class providing operations for Google Cloud Bigquery.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This abstract class is provided to enable testability while permitting
+    /// additional operations to be added in the future. See <see cref="Create(GoogleCredential)"/>
+    /// and <see cref="CreateAsync(GoogleCredential)"/> to construct instances; alternatively, you can
+    /// construct a <see cref="BigqueryClientImpl"/> directly from a <see cref="BigqueryService"/>.
+    /// </para>
+    /// <para>
+    /// All instance methods declared in this class are virtual, with an implementation which simply
+    /// throws <see cref="NotImplementedException"/>. All these methods are overridden in <see cref="BigqueryClientImpl"/>.
+    /// </para>
+    /// </remarks>
+    public abstract partial class BigqueryClient
+    {
+        /// <summary>
+        /// The underlying Bigquery service object used by this client.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="BigqueryClient"/> class only provides convenience operations built on top of
+        /// an existing service object. Any more complex operations which are not supported by this wrapper may wish
+        /// to use the same service object as the wrapper, in order to take advantage of its configuration (for authentication,
+        /// application naming etc).
+        /// </remarks>
+        public virtual BigqueryService Service { get; }
```

Comment:
In my opinion, this getter should throw a NotImplementedException.
This is what happens in gRPC gen'ed code.